src/Index/utils.js

```diff
@@ -32,40 +92,41 @@ export const loadHomepageChartDataFromDatasets = async (callback) => {
   const fileNodes = store.getState().submission.file_nodes;
   const nodesForIndexChart = homepageChartNodes.map(item => item.node);
   const nodesToRequest = _.union(fileNodes, nodesForIndexChart);
-  const url = `${datasetUrl}?nodes=${nodesToRequest.join(',')}`;
-
-  fetchWithCreds({
-    path: url,
-  }).then((res) => {
-    switch (res.status) {
-    case 200:
-      updateRedux(res.data);
-      if (callback) {
-        callback(resultStatus);
-      }
-      break;
-    case 404:
+  const requestUrls = getChunkedPeregrineRequestUrls(nodesToRequest);
+
+  const fullResultPromise = [];
+  for (let k = 0; k < requestUrls.length; k += 1) {
+    const chunkResult = makePeregrineRequestForNode(requestUrls[k]);
+    fullResultPromise.push(chunkResult);
+  }
+
+  const fullResult = await Promise.all(fullResultPromise);
+
+  const queryFailure = fullResult.some(element => element[1] !== 200);
+  const query401 = fullResult.some(element => element[1] === 401);
+  const query403 = fullResult.some(element => element[1] === 403);
+  const query404 = fullResult.some(element => element[1] === 404);
+
+  if (queryFailure) {
+    if (query404) {
       // Shouldn't happen, this means peregrine datasets endpoint not enabled
       console.error(`REST endpoint ${datasetUrl} not enabled in Peregrine yet.`);
+    }
+    if (query401 || query403 || query404) {
       resultStatus.needLogin = true;
       if (callback) {
         callback(resultStatus);
       }
-      break;
-    case 401:
-    case 403:
-      resultStatus.needLogin = true;
-      if (callback) {
-        callback(resultStatus);
-      }
-      break;
-    default:
-      break;
     }
-  })
-    .catch((err) => {
-      console.log(err);
-    });
+    return;
```

Comment:
also print something for other errors?

Reply:
Are you sure that's desirable? The original functionality did not print something for the other errors. If a query comes back with status code 500, I don't think a console print stating "The homepage chart node query came back with status code 500" would be desirable, or consistent with our usual error handling style in this repo.

Reply:
i don't have any strong feeling about it. but i think it would be more helpful to know what went wrong, plus we are already printing some error messages in other functions